### PR TITLE
Timer

### PR DESCRIPTION
**Pull request by dotgourav**
Adds timer functionality with start, stop, resume and pause.

### CODE REVIEW

File: app/modules/timer/controllers/timer.controller.js [hunk 1]
```diff
@@ -9,23 +9,106 @@
...
 
   /**
    * @namespace TimerController
    */
-  function TimerController($location, $scope, Timer) {
-    var vm = this;
+  function TimerController($location, $scope, $interval, Timer) {
+    let vm = this;
+    let timerInterval = null;
+    let minutes = 25;
```
Comment (Nikhilkapoor20) [inline app/modules/timer/controllers/timer.controller.js]: we should add this to service.

File: app/modules/timer/controllers/timer.controller.js [hunk 2]
```diff
@@ -9,23 +9,106 @@
...
+    let seconds = 0;
```
Comment (Nikhilkapoor20) [inline app/modules/timer/controllers/timer.controller.js]: we should add this to service.

File: app/modules/timer/controllers/timer.controller.js [hunk 3]
```diff
@@ -9,23 +9,106 @@
...
 
+    vm.displayTimer = '' + minutes + ' : ' + seconds + 's';
+    vm.isTimerActive = false;
+    vm.isTimerPaused = false;
     vm.startTimer = startTimer;
+    vm.pauseTimer = pauseTimer;
+    vm.resumeTimer = resumeTimer;
+    vm.stopTimer = stopTimer;
+    
+    let setDisplayTimer = function(minutes, seconds) {
```
Comment (Nikhilkapoor20) [inline app/modules/timer/controllers/timer.controller.js]: Comment missing on function

File: app/modules/timer/controllers/timer.controller.js [hunk 4]
```diff
@@ -9,23 +9,106 @@
...
+      let minutesString = minutes < 10 ? '0' + minutes : '' + minutes;
+      let secondsString = seconds < 10 ? '0' + seconds + 's' : '' + seconds + 's';
+      vm.displayTimer = minutesString + ' : ' + secondsString;
+    };
+
+    let clock = function() {
```
Comment (Nikhilkapoor20) [inline app/modules/timer/controllers/timer.controller.js]: Comment missing on function

File: app/modules/timer/controllers/timer.controller.js [hunk 5]
```diff
@@ -9,23 +9,106 @@
...
+          seconds = 60;
+          minutes = minutes - 1;
+        }
+        seconds = seconds - 1;
+        setDisplayTimer(minutes, seconds);
+      }
+    };
+
+    let resetTimer = function() {
+      minutes = 10;
```
Comment (Nikhilkapoor20) [inline app/modules/timer/controllers/timer.controller.js]: why minutes = 10 ?

**Review by Nikhilkapoor20 — CHANGES_REQUESTED** [review]
check comments

File: app/modules/timer/controllers/timer.controller.js (hunk 5 above)
Comment (dotgourav) [reply app/modules/timer/controllers/timer.controller.js]: This is supposed to be 25.

File: app/modules/timer/controllers/timer.controller.js (hunk 1 above)
Comment (Nikhilkapoor20) [reply app/modules/timer/controllers/timer.controller.js]: on stop it is not resetting to 00 : 00

File: app/modules/timer/controllers/timer.controller.js [hunk 6]
```diff
@@ -9,23 +9,106 @@
...
+      let secondsString = seconds < 10 ? '0' + seconds + 's' : '' + seconds + 's';
+      vm.displayTimer = minutesString + ' : ' + secondsString;
+    };
+
+    let clock = function() {
+      if(minutes === 0) {
+        cancelInterval(timerInterval);
+        vm.isTimerActive = false;
+      } else {
+        if(seconds === 0) {
```
Comment (Nikhilkapoor20) [inline app/modules/timer/controllers/timer.controller.js]: space after if and function is missing

File: app/modules/timer/controllers/timer.controller.js [hunk 7]
```diff
@@ -9,23 +9,106 @@
...
+    vm.stopTimer = stopTimer;
+    
+    let setDisplayTimer = function(minutes, seconds) {
+      let minutesString = minutes < 10 ? '0' + minutes : '' + minutes;
+      let secondsString = seconds < 10 ? '0' + seconds + 's' : '' + seconds + 's';
+      vm.displayTimer = minutesString + ' : ' + secondsString;
+    };
+
+    let clock = function() {
+      if(minutes === 0) {
```
Comment (Nikhilkapoor20) [inline app/modules/timer/controllers/timer.controller.js]: space after if and function is missing

File: app/modules/timer/controllers/timer.controller.js [hunk 8]
```diff
@@ -9,23 +9,106 @@
...
+        if(seconds === 0) {
+          seconds = 60;
+          minutes = minutes - 1;
+        }
+        seconds = seconds - 1;
+        setDisplayTimer(minutes, seconds);
+      }
+    };
+
+    let resetTimer = function() {
```
Comment (Nikhilkapoor20) [inline app/modules/timer/controllers/timer.controller.js]: space after if and function is missing

File: app/modules/timer/controllers/timer.controller.js (hunk 5 above)
Comment (Nikhilkapoor20) [inline app/modules/timer/controllers/timer.controller.js]: reset function should set time back to 25:00 s , please consider this case too

**Review by dotgourav — COMMENTED** [review]
Merging this as I’m changing the code structure inthe controller.

**Pull request merged**